test/runtests.jl

```diff
@@ -220,11 +221,195 @@ end
+    @test refcount(pa2) === refcount(pa1)
+    @test refcount(pa2)[] == 2
+
+    pa1 = PooledArray([1, 2, 3])
+    pa2 = similar(pa1, 4)
+    copyto!(pa2, 1, view(pa1, [1, 1]), 1, 2)
+    @test refpool(pa2) === refpool(pa1)
+    @test invrefpool(pa2) === invrefpool(pa1)
+    @test refcount(pa2) === refcount(pa1)
+    @test refcount(pa2)[] == 2
+    copyto!(pa2, 3, view(pa1, [1, 1]), 1, 2)
+    @test refpool(pa2) === refpool(pa1)
+    @test invrefpool(pa2) === invrefpool(pa1)
+    @test refcount(pa2) === refcount(pa1)
+    @test refcount(pa2)[] == 2
+    @test pa2 == [1, 1, 1, 1]
+
+    pa1 = PooledArray([1, 2, 3])
+    pa2 = similar(pa1, Float64, 3)
+    copyto!(pa2, 1, pa1, 1, 3)
+    @test refpool(pa2) !== refpool(pa1)
+    @test invrefpool(pa2) !== invrefpool(pa1)
+    @test refcount(pa2) !== refcount(pa1)
+    @test refcount(pa1)[] == 1
+    @test refcount(pa2)[] == 1
+    @test pa2 == [1, 2, 3]
+
+    pa1 = PooledArray([1, 2, 3])
+    pa2 = similar(pa1, 3)
+    pa2 .= 1
+    copyto!(pa2, 1, pa1, 1, 3)
+    @test refpool(pa2) !== refpool(pa1)
+    @test invrefpool(pa2) !== invrefpool(pa1)
+    @test refcount(pa2) !== refcount(pa1)
+    @test refcount(pa1)[] == 1
+    @test refcount(pa2)[] == 1
+    @test pa2 == [1, 2, 3]
+end
+
+@testset "reverse" begin
+    pa1 = PooledArray([1, 2, 3])
+    pa2 = reverse(pa1)
+    pa3 = reverse(pa2)
+    @test pa2 == [3, 2, 1]
+    @test pa3 == pa1
+    @test refpool(pa1) === refpool(pa2) === refpool(pa3)
+    @test invrefpool(pa1) === invrefpool(pa2) === invrefpool(pa3)
+    @test refcount(pa1) === refcount(pa2) === refcount(pa3)
+    @test refcount(pa1)[] == 3
+end
 
+@testset "convert" begin
+    pa1 = PooledArray([1, 2, 3])
+    @test convert(PooledArray, pa1) === pa1
+    @test eltype(convert(PooledArray{Float64}, pa1)) === Float64
+    @test convert(PooledArray{Int, UInt64, 1}, pa1) isa PooledArray{Int,UInt64,1,Array{UInt64,1}}
+end
+
+@testset "indexing" begin
+    pa = PooledArray([1 2; 3 4])
+    @test pa[2, 2, 1] == 4
+    @test pa[2, 2] == 4
+    @test pa[big(2), 2] == 4
+    pav = view(pa, :, :)
+    @test pav[2, 2, 1] == 4
+    @test pav[2, 2] == 4
+    @test pav[big(2), 2] == 4
+
+    @test refcount(pa)[] == 1
+    pa2 = pa[[true, false, false, true]]
+    @test pa2 == [1, 4]
+    @test refpool(pa) === refpool(pa2)
+    @test invrefpool(pa) === invrefpool(pa2)
+    @test refcount(pa) === refcount(pa2)
+    @test refcount(pa)[] == 2
+    pa3 = pav[[true, false, false, true]]
+    @test pa3 == [1, 4]
+    @test refpool(pa) === refpool(pa3)
+    @test invrefpool(pa) === invrefpool(pa3)
+    @test refcount(pa) === refcount(pa3)
+    @test refcount(pa)[] == 3
+end
+
+@testset "isassigned" begin
+    pa1 = PooledArray(["a"])
+    pa2 = similar(pa1, 2)
+    pa2v = view(pa2, 1)
+    @test !isassigned(pa2, 1)
+    @test !isassigned(pa2v)
+end
+
+@testset "setindex!" begin
+    pa = PooledArray([1 2; 3 4])
+    pa[1, 1] = 10
+    @test pa == [10 2; 3 4]
+    @test [pa pa] == [10 2 10 2; 3 4 3 4]
```

Comment:
This isn't `setindex!`. Move it to another test set?

Reply:
It is `setindex!` test actually. Only you do not see it. `hcat` uses `setindex!`.

This is how this test works on 1.1 release on PooledArrays.jl:
```
julia> pa = PooledArray([1, 2, 3])
3-element PooledVector{Int64, UInt32, Vector{UInt32}}:
 1
 2
 3

julia> [pa pa]
ERROR: setindex! not defined for PooledMatrix{Int64, UInt32, Matrix{UInt32}}
```

Exactly because of such problems I opt for making `PooledArray` support `IndexLinear` only (this fails because on 1.1 `PooledArray` is `IndexCartesian`).